src/components/SearchBox/SearchBox.js

```diff
@@ -72,13 +73,18 @@ class SearchBox extends Component {
 
     if (searchAsYouType) {
       refine(query);
-    } else {
-      this.setState({ query });
     }
+    this.setState({ query });
 
     onChange(event);
   };
 
+  componentWillReceiveProps(nextProps) {
```

Comment:
i researched, and it doesn't seem like this method is deprecated in preact.

Reply:
yes, I found the same.
Also I tried using getDerivedPropsFromState. It worked in the browser, but it seems the renderer we use in the tests does not fire it, and a lot of tests failed as a result.